ush/wave_tar.sh

```diff
@@ -98,21 +98,27 @@ source "${USHgfs}/preamble.sh"
   echo '   Making tar file ...'
   set_trace
 
-  count=0
   countMAX=5
   tardone='no'
-
-  while [ "$count" -lt "$countMAX" ] && [ "$tardone" = 'no' ]
+  sleep_interval=10
+  
+  while [[ "${tardone}" = "no" ]]
   do
     
     nf=$(ls | awk '/'$ID.*.$filext'/ {a++} END {print a}')
     nbm2=$(( $nb - 2 ))
-    if [ $nf -ge $nbm2 ]
-    then 
-      tar -cf $ID.$cycle.${type}_tar ./$ID.*.$filext
+    if [[ "${nf}" -ge "${nbm2}" ]]
+    then
+
+      filename="${ID}.${cycle}.${type}_tar" 
```

Comment:
Adding filename here is a great addition, is there a reason we don't use this in other places too?  Lastly, I'm concerned about the change in order here.  Shouldn't we create the tar-file before we wait to see if it exists? I might not have looked at the logic long enough here, but this seems out of order now.

Reply:
Yup, the tar creation needs to be before the wait line. But I'm not really sure of the point of how it was set up originally anyway. Shouldn't be waiting on the existence of the files that are going _into_ the tar?

Reply:
Agreed, it looks a bit odd which is why i was thinking maybe @HenryWinterbottom-NOAA was seeing something I wasn't seeing after a quick look.  What I suspect is happening here is that we're making sure the tar file exists and is a certain age before creating a tar.gz below -- but again a quick look and would have to do a deeper dive to refresh my memory or understand why things are this way.

Reply:
Yeah. For now, the task is just to replicate the functionality of what was there before (so make the tarball first). When these get refactored, we will need to look at changing it (this job looks ripe for the wxflow `FileHandler`).

Reply:
Is this resolved or am I missing something here?

Reply:
@HenryWinterbottom-NOAA not resolved. You need to move the line with the `tar` command above where you check for the resultant file.

Reply:
Done.